chained/tlsfrag.go

```diff
@@ -32,20 +34,41 @@ func makeFragFunc(pc *config.ProxyConfig) (tlsfrag.FragFunc, bool) {
 	}
 
 	// fragStr should be of the form <frag func> or <frag func>:<func config>.
-	// Example: "index:3"
+	// Example: "index:3" or "regex:foo.*" or "indexOf:test"
 	funcType, cfg, hasCfg := strings.Cut(fragStr, ":")
+	if !hasCfg {
+		log.Error("tlsfrag: missing config specifier")
+		return nil, false
+	}
 	switch funcType {
 	case "index":
-		if !hasCfg {
-			log.Error("tlsfrag: missing index specifier")
-			return nil, false
-		}
-
 		index, err := strconv.Atoi(cfg)
 		if err != nil {
 			log.Errorf("tlsfrag: bad index specifier: %v", err)
+			return nil, false
 		}
 		return func(_ []byte) int { return index }, true
+	case "regex":
+		regex, err := regexp.Compile(cfg)
+		if err != nil {
+			log.Errorf("tlsfrag: bad regex specifier: %v", err)
+			return nil, false
+		}
+		return func(b []byte) int {
+			loc := regex.FindIndex(b)
+			if loc == nil {
+				return 0
+			}
+			return loc[0]
+		}, true
+	case "indexOf":
+		return func(b []byte) int {
+			index := bytes.Index(b, []byte(cfg))
```

Comment:
This has the same problem as the regular expression function, but could also be solved by treating the input as a hex-encoded string.

Reply:
OK added the hex encoding with just dividing by two to get the index -- good idea!

Reply:
Oh I ditched the indexOf btw because you can just use regex with a plain string to achieve the same thing (i.e. bytes.Index is the same as regex.FindIndex with a naked string.